pgex/examples/platformer_collisions/main.py

```diff
@@ -0,0 +1,125 @@
+"""
+This file is a part of the 'Pygame Examples (pgex)' source code.
+The source code is distributed under the MIT license.
+"""
+
+import asyncio
+
+import pygame
+
+from . import common
+
+
+def get_colliding_tiles(tiles: list, player_rect: pygame.Rect) -> list:
+    """
+    Returns a list of tiles the player is currently colliding with
+    """
+    return_tiles = []
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            return_tiles.append(tile)
+
+    return tiles
+
+
+def calculate_rect(
+    movement: list, player_rect: pygame.Rect, map_tiles: list
+) -> pygame.Rect:
+    """
+    Calculates the Rect of the player based on their movement and the surrounding tiles
+    """
+    player_rect.x += movement[0]
+    tiles = get_colliding_tiles(map_tiles, player_rect)
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            if movement[0] > 0:
+                player_rect.right = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).left
+            if movement[0] < 0:
+                player_rect.left = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).right
+
+    common.is_on_ground = False
+    player_rect.y += movement[1]
+    tiles = get_colliding_tiles(map_tiles, player_rect)
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            if movement[1] > 0:
+                player_rect.bottom = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).top
+                common.is_on_ground = True
+                common.y_velocity = 3
+            if movement[1] < 0:
+                player_rect.top = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).bottom
+
+    return player_rect
+
+
+def player_movement(key_presses: dict, player_rect: pygame.Rect) -> pygame.Rect:
+    """
+    Handles all code realting to the movement of the player
+    """
+
+    player_movement = [0, common.y_velocity]
+    if key_presses["a"]:
+        player_movement[0] -= 4
+    if key_presses["d"]:
+        player_movement[0] += 4
+
+    if common.y_velocity < 10:
+        common.y_velocity += 0.5
+
+    player_rect = calculate_rect(player_movement, player_rect, common.map_data["map"])
+    return player_rect
+
+
+def render_map(display: pygame.Surface, tiles: list) -> None:
+    """
+    Renders the games tiles
+    """
+
+    for tile in tiles:
+        pygame.draw.rect(display, (60, 255, 100), tile)
```

Comment:
Out of context I don't know what color is bound to `(60, 255, 100)`, it could be extracted as a constant somewhere.
Or you could also rely on predefined colors by pygame: https://github.com/pygame/pygame/blob/main/src_py/colordict.py
(example: `pygame.Color("yellow1")`)

Reply:
Makes sense. If you need to put the background color in different places, then having identifiers per "place on the UI" is definitely useful. 

But you can still have what I said.

Example:

`RED = (0, 0, 255)
 GREEN = (0, 255, 0) 
 ...

BG_COLOR = RED`

Or using the predefined named colors provided by pygame.

But I agree to what you are going to say: it would be over-engineering for such a small project.